### ISSUE TITLE
Idle disconnect/timeout issue (3.6.1)

### ISSUE THREAD

**Issue by gmr** [rabbitmq/rabbitmq-auth-backend-ldap#30]
Looks like the socket connection for LDAP was closed (remotely?) on a server that sat idle over night and the `reuse` call doesn't handle the case of a closed socket when it expects it to be open. This results in an exception in the management UI when logging in, in this edge case:

```
Error in process <0.7188.0> on node 'rabbit@edeliv-rabbit1' with exit value:
{{nocatch,{gen_tcp_error,closed}},
 [{eldap,send_the_LDAPMessage,3,[{file,"eldap.erl"},{line,991}]},
  {eldap,do_unbind,1,[{file,"eldap.erl"},{line,955}]},
  {eldap,loop,2,[{file,"eldap.erl"},{line,567}]}]}

=ERROR REPORT==== 6-Apr-2016::12:29:08 ===
webmachine error: path="/api/whoami"
{exit,{{{nocatch,{gen_tcp_error,closed}},
        [{eldap,send_the_LDAPMessage,3,[{file,"eldap.erl"},{line,991}]},
         {eldap,do_unbind,1,[{file,"eldap.erl"},{line,955}]},
         {eldap,loop,2,[{file,"eldap.erl"},{line,567}]}]},
       {gen_server2,call,
                    [<0.5123.0>,
                     {submit,#Fun<rabbit_auth_backend_ldap.9.105034196>,
                             <0.20918.14>,reuse},
                     infinity]}},
      [{gen_server2,call,3,[]},
       {rabbit_federation_status,status,0,[]},
       {lists,foldl,3,[]},
       {rabbit_federation_status,status,0,[]}]}
```

**Comment by johnfoldager** [comment]
We see this too in a setup where RabbitMQ 3.6.2 RC1 connects through a Microsoft Azure Load Balancer to a LDAP server.

**Comment by hairyhum** [comment]
Fixed by workaround in #42
